src/build/app/settings.rs

```diff
@@ -904,6 +907,36 @@ pub enum AppSettings {
     /// [``]: ./struct..html
     SubcommandRequiredElseHelp,
 
+    /// Specifies that the help subcommand should print the long form help message (if possible), instead of the short
+    /// form.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use clap::{App, AppSettings};
+    /// let m = App::new("myprog")
+    ///     .setting(AppSettings::SubcommandHelpShowsLongForm)
+    ///     .subcommand(App::new("test")
+    ///         .after_help("short form help msg")
+    ///         .after_long_help("long form help msg"))
+    ///     .get_matches();
+    /// // `myprog help test` will display the auto-generated help, as well as
+    /// // "long form help msg"
+    /// ```
+    ///
+    /// ```rust
+    /// # use clap::{App, AppSettings};
+    /// let m = App::new("myprog")
+    ///     .setting(AppSettings::SubcommandHelpShowsLongForm)
+    ///     .subcommand(App::new("test")
+    ///         .before_help("short form help msg")
+    ///         .before_long_help("long form help msg"))
+    ///     .get_matches();
+    /// // `myprog help test` will display "long form help msg", followed by auto-generated help
+    /// ```
+    /// [``]: ./struct..html
+    SubcommandHelpShowsLongForm,
```

Comment:
Let's rename this to `UseLongFormatForHelpSubcommand`. (need to rename all the other stuff too)

Reply:
Refacroted `SubcommandHelpShowsLongForm` to be `UseLongFormatForHelpSubcommand` instead